app/src/app/index.jsx

```diff
@@ -157,24 +159,59 @@ const MiscRouter = () => ([
     <Route component={NotFoundPage} key="NotFoundPage" />,
 ])
 
+type Props = {
+    children: any,
+}
+
+type State = {}
+
+// This is a way to handle all errors in the app.
+class ErrorBoundary extends Component<Props, State> {
+    componentDidMount() {
+        if (isProduction()) {
+            console.log('prodigy')
+            Sentry.init({
+                dsn: process.env.SENTRY_DSN,
+                release: process.env.VERSION,
+                environment: process.env.SENTRY_ENVIRONMENT,
+                debug: true,
+            })
+        }
+    }
+
+    componentDidCatch(error, errorInfo) {
+        if (isProduction()) {
+            Sentry.captureException(error, {
+                extra: errorInfo,
+            })
```

Comment:
I know you're just testing but just in case you're wondering where the `extra` got to, I don't think this is how to attach `extra` any more, this is old raven syntax, need to `withScope` and `addExtra` like so: https://github.com/streamr-dev/streamr-platform/blob/development/app/src/analytics.js#L65-L73

Reply:
Good point though, we could probably do with a top level error handler

Reply:
ahh roger that, cheers 👍